cmd/helm/flags.go

```diff
@@ -112,33 +115,82 @@ func (o *outputValue) Set(s string) error {
 }
 
 func bindPostRenderFlag(cmd *cobra.Command, varRef *postrender.PostRenderer) {
-	cmd.Flags().Var(&postRenderer{varRef}, postRenderFlag, "the path to an executable to be used for post rendering. If it exists in $PATH, the binary will be used, otherwise it will try to look for the executable at the given path")
+	p := &postRendererOptions{varRef, "", []string{}}
+	cmd.Flags().Var(&postRendererString{p}, postRenderFlag, "the path to an executable to be used for post rendering. If it exists in $PATH, the binary will be used, otherwise it will try to look for the executable at the given path")
+	cmd.Flags().Var(&postRendererArgsSlice{p}, postRenderArgsFlag, "the args to an executable to be used for post rendering.  (can specify multiple)")
+}
+
+type postRendererOptions struct {
+	renderer   *postrender.PostRenderer
+	binaryPath string
+	args       []string
 }
 
-type postRenderer struct {
-	renderer *postrender.PostRenderer
+type postRendererString struct {
+	options *postRendererOptions
 }
 
-func (p postRenderer) String() string {
-	return "exec"
+func (p *postRendererString) String() string {
+	return p.options.binaryPath
 }
 
-func (p postRenderer) Type() string {
-	return "postrenderer"
+func (p *postRendererString) Type() string {
+	return "postRendererString"
 }
 
-func (p postRenderer) Set(s string) error {
-	if s == "" {
+func (p *postRendererString) Set(val string) error {
+	if val == "" {
 		return nil
 	}
-	pr, err := postrender.NewExec(s)
+	p.options.binaryPath = val
+	pr, err := postrender.NewExec(p.options.binaryPath, p.options.args...)
 	if err != nil {
 		return err
 	}
-	*p.renderer = pr
+	*p.options.renderer = pr
 	return nil
 }
 
+type postRendererArgsSlice struct {
+	options *postRendererOptions
+}
+
+func (p *postRendererArgsSlice) String() string {
+	return "[" + strings.Join(p.options.args, ",") + "]"
+}
+
+func (p *postRendererArgsSlice) Type() string {
+	return "postRendererArgsSlice"
+}
+
+func (p *postRendererArgsSlice) Set(val string) error {
+	if val == "" || p.options.binaryPath == "" {
```

Comment:
This is a bit of a corner case, but you had it better in your previous version.
Even when `p.options.binaryPath == ""` we should still call `p.options.args = append(p.options.args, val)`.  This allows to use the `--post-renderer-args` before the `--post-renderer` flag on the command-line.
For example, if I run:
```
$ bin/helm template test bitnami/nginx --post-renderer-args arg1  --post-renderer echo --post-renderer-args arg2
arg2
```
Notice that arg1 is ignored because it was specified before `--post-renderer`

Reply:
got! .  i will fix all of these.